app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -61,4 +82,50 @@ class SubscriptionService @Inject()(agentSubscriptionConnector: AgentSubscriptio
       case e => throw e
     }
   }
+
+  def completePartialSubscription(details: CompletePartialSubscriptionBody)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[(Boolean, Option[Arn])] =
+    for {
+      subscriptionProgress <- getSubscriptionStatus(details.utr, details.knownFacts.postcode) //FixMe APB-2805 this check should happen previously and can be stored in the session
+      isPartiallySubscribed = subscriptionProgress.state == SubscriptionState.IsOnlySubscribedInETMP
+      (wasPartiallySubscribed, optSucceededAllocatingArn) <- isPartiallySubscribed match {
+                                                              case true =>
+                                                                agentSubscriptionConnector
+                                                                  .completePartialSubscription(details)
+                                                                  .map(arn => (true, Some.apply(arn)))
+                                                                  .recover {
+                                                                    case e: Upstream4xxResponse
+                                                                        if Seq(Status.FORBIDDEN, Status.CONFLICT) contains e.upstreamResponseCode =>
+                                                                      (true, None)
```

Comment:
Here Status.FORBIDDEN eventually leads us to showAlreadySubscribed page. Is that the correct behaviour? The backend returns FORBIDDEN when there is not ETMP record/ postcodes don't match/ isAnASAgent flag on etmp record is false. I think we should also log these exceptions.

Reply:
this is different logic, it tells (wasPartiallySubscribed, optSucceededAllocatingArn) and you need Some(Arn) to be subscribed, this only happens if you are if !reg.isSubscribedToAgentServices && reg.isSubscribedToETMP
and your status is eligible for partialSubscription since SubscriptionState.IsOnlySubscribedInETMP,
you will land on showAlreadySubscribed if reg.isSubscribedToAgentServices && reg.isSubscribedToETMP which makes you SubscriptionState.IsSubscribedToAgentServices